website/src/release.md

```diff
@@ -231,7 +231,7 @@ https://downloads.apache.org/iceberg/KEYS
 
 Git tag for the release:
 
-https://github.com/apache/iceberg-rust/releases/tag/${release_version}
+https://github.com/apache/iceberg-rust/releases/tag/v${iceberg_version}
```

Comment:
already fixed lower down the same file
https://github.com/apache/iceberg-rust/blob/1d217d4943bcce0ec97bb135fba10cda14da6b79/website/src/release.md#L389